benchmark/custom_csv_formater.hpp

```diff
@@ -154,16 +157,34 @@ void customCSVReporter::PrintRunData(const Run& run)
     //get the name of the engine and distribution:
 
     std::string temp = run.benchmark_name();
+
+    std::string deviceName = std::string(temp.begin(), temp.begin() + temp.find("<"));
+
     temp.erase(0, temp.find("<") + 1);
 
     std::string engineName = std::string(temp.begin(), temp.begin() + temp.find(","));
 
     temp.erase(0, engineName.size() + 1);
-    temp.erase(0, temp.find(",") + 1);
+
+    std::string mode = "default";
+
+    if(deviceName != "device_kernel")
+    {
+        mode = std::string(temp.begin(), temp.begin() + temp.find(','));
+        temp.erase(0, temp.find(",") + 1);
+    }
     std::string disName = std::string(temp.begin(), temp.begin() + temp.find(">"));
 
-    Out << engineName << ",";
-    Out << disName << ",";
+    std::string lambda = "";
+
+    size_t ePos = disName.find("=");
+    if(ePos <= disName.size())
+    {
+        lambda = std::string(disName.begin() + (ePos + 1), disName.end() - 1);
```

Comment:
I think the string parsing in this function is fine for this task, because we're expecting things to always be present and always be in the same order. However, just for future reference, using a regular expression with capture groups can be really useful in situations where you need to pull out a bunch of sub-strings like this. You can create a regex pattern that contains all the delimiters ('<', ','), apply it to the original temp string, and then iterate through the captured text in a single loop.

Reply:
okay thank you, sounds good